pkgs/applications/terminal-emulators/mlterm/default.nix

```diff
@@ -62,20 +165,13 @@ stdenv.mkDerivation rec {
   '';
 
   configureFlags = [
-    "--with-imagelib=gdk-pixbuf" #or mlimgloader depending on your bugs of choice
-    "--with-type-engines=cairo,xft,xcore"
-    "--with-gtk=3.0"
-    "--enable-ind" #indic scripts
-    "--enable-fribidi" #bidi scripts
-    "--with-tools=mlclient,mlconfig,mlcc,mlterm-menu,mlimgloader,registobmp,mlfc"
-    #mlterm-menu and mlconfig depend on enabling gnome.at-spi2-core
-    #and configuring ~/.mlterm/key correctly.
-  ] ++ lib.optionals (!stdenv.isDarwin) [
-    "--with-x=yes"
-    "--with-gui=xlib,fb"
-    "--enable-m17nlib" #character encodings
-  ] ++ lib.optionals stdenv.isDarwin [
-    "--with-gui=quartz"
+    (withFeaturesList "type-engines" enableTypeEngines)
+    (withFeaturesList "tools" enableTools)
+    (withFeaturesList "gui" enableGuis)
```

Comment:
This might be a bit overkill/too abstract.

I didn't know lib.withFeature before though, so perhaps it's more obvious if you do.